### ISSUE TITLE
Question: How can we invalidate all keys that start with something.

### ISSUE THREAD

**Issue by RodolfoSilva** [D-James-GH/cached_query#7]
Does the `invalidate` method invalidate nested keys?
For example:

`keys: ['posts']`
`keys: ['posts', 1234]`

If I invalidate the `'posts'` key, will all the keys that start with the `['posts']` brace be invalidated?
In ReactQuery this will invalidate all keys starting with `'posts'`.

**Comment by D-James-GH** [comment]
Hi, 
Unlike React Query passing keys will match the exact keys. However you can get the same effect with one of two ways: 
1. Using a filterFn
```dart
CachedQuery.instance.invalidateCache(
  filterFn: (unencodedKey, key) => unencodedKey is List && unencodedKey.first == "posts",
);
```
A filterFn just needs to return `true` or `false`. Invalidate queries will invalidate any of the queries that match the filter.

The `unencodedKey` is the key you originally passed to it eg `["posts", 123]`. The key passed to the filterFn is the json_encoded version of the key.

2. Find a query using `whereQuery` then invalidate.
```dart
final queries = CachedQuery.instance.whereQuery((query) => query.key.contains("posts"));
queries.forEach((q) => q.invalidateQuery());
```

Please also note that `invalidateQuery` does not function exactly the same as React Query. In CachedQuery the invalidate function will only set the query as "stale". This means it is guaranteed to be fetched next time the `queryFn` is called. 

What you might be looking for is `refetch` or `refetchQueries`. Which will refetch the queries immediately without waiting for a `queryFn` to be called.

**Comment by RodolfoSilva** [comment]
Solved by #6
